libfreerdp/codec/h264_mediacodec.c

```diff
@@ -0,0 +1,516 @@
+            if (outputBufferId >= 0)
+            {
+                sys->currentOutputBufferIndex = outputBufferId;
+
+                uint8_t* outputBuffer;
+                outputBuffer = sys->fnAMediaCodec_getOutputBuffer(sys->decoder, outputBufferId, &outputBufferSize);
+                sys->currentOutputBufferIndex = outputBufferId;
+
+                if (outputBufferSize != (sys->width * sys->height + ((sys->width + 1) / 2) * ((sys->height + 1) / 2) * 2))
+                {
+                    WLog_Print(h264->log, WLOG_ERROR, "Error unexpected output buffer size %d", outputBufferSize);
+                    return -1;
+                }
+
+                // TODO: work with AImageReader and get AImage object instead of COLOR_FormatYUV420Planar buffer.
+                iStride[0] = sys->width;
+                iStride[1] = (sys->width + 1) / 2;
+                iStride[2] = (sys->width + 1) / 2;
+                pYUVData[0] = outputBuffer;
+                pYUVData[1] = outputBuffer + iStride[0] * sys->height;
+                pYUVData[2] = outputBuffer + iStride[0] * sys->height + iStride[1] * ((sys->height + 1) / 2);
+                break;
+            }
+            else if (outputBufferId ==  AMEDIACODEC_INFO_OUTPUT_FORMAT_CHANGED)
+            {
+                AMediaFormat* outputFormat;
+                outputFormat = sys->fnAMediaCodec_getOutputFormat(sys->decoder);
+                if (outputFormat == NULL)
+                {
+                    WLog_Print(h264->log, WLOG_ERROR, "AMediaCodec_getOutputFormat failed");
+                    return -1;
+                }
+                set_mediacodec_format(h264, &sys->outputFormat, outputFormat);
+
+                media_format = sys->fnAMediaFormat_toString(sys->outputFormat);
+                if (media_format == NULL)
+                {
+                    WLog_Print(h264->log, WLOG_ERROR, "AMediaFormat_toString failed");
+                    return -1;
+                }
+            }
+            else if (outputBufferId ==  AMEDIACODEC_INFO_TRY_AGAIN_LATER)
+            {
+                WLog_Print(h264->log, WLOG_WARN, "AMediaCodec_dequeueOutputBuffer need to try again later");
+                // TODO: sleep?
+            }
+            else if (outputBufferId == AMEDIACODEC_INFO_OUTPUT_BUFFERS_CHANGED)
+            {
+                WLog_Print(h264->log, WLOG_WARN, "AMediaCodec_dequeueOutputBuffer returned deprecated value AMEDIACODEC_INFO_OUTPUT_BUFFERS_CHANGED, ignoring");
+            }
+            else
+            {
+                WLog_Print(h264->log, WLOG_ERROR, "AMediaCodec_dequeueOutputBuffer returned unknown value [%d]", outputBufferId);
+                return -1;
+            }
+        }
+
+        break;
+    }
+
+    return 1;
+}
+
+static void mediacodec_uninit(H264_CONTEXT* h264)
+{
+    H264_CONTEXT_MEDIACODEC* sys = (H264_CONTEXT_MEDIACODEC*)h264->pSystemData;
```

Comment:
Please `WINPR_ASSERT(h264)` and others before accessing them without `NULL` check.